core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -182,6 +247,186 @@ private static FunctionResolver log2() {
         singleArgumentFunction(BuiltinFunctionName.LOG2.getName(), v -> Math.log(v) / Math.log(2)));
   }
 
+  /**
+   * Definition of mod(x, y) function.
+   * Calculate the remainder of x divided by y
+   * The supported signature of mod function is
+   * (INTEGER, INTEGER) -> INTEGER
+   * (LONG, LONG) -> LONG
+   * (FLOAT, FLOAT) -> FLOAT
+   * (DOUBLE, DOUBLE) -> DOUBLE
+   */
+  private static FunctionResolver mod() {
```

Comment:
does mod support LONG and FLOAT?

Reply:
MySQL seems support float and long? https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_mod

@chloe-zh Btw, does Java native '%' cover this case in MySQL doc above? "MOD(N,0) returns NULL."

Reply:
LONG type should be good as says in MySQL manual (as BIGINT), but it doesn't mention about FLOAT.

Reply:
@dai-chen Good point! yes currently it also throws arithmetic exception. Will change its behavior to return NULL instead.